### ISSUE TITLE
Use rename_all attribute

### ISSUE THREAD

**Issue by dtolnay** [evestera/json_typegen#1]
Amazing work!

I noticed this behavior:

```json
{
  "aaAaaAaa": 0,
  "bbBbbBbb": 0,
  "ccCccCcc": 0
}
```

```rust
#[derive(Default, Debug, Clone, Serialize, Deserialize)]
struct ZalandoArticle {
    #[serde(rename = "aaAaaAaa")]
    aa_aaa_aaa: i64,
    #[serde(rename = "bbBbbBbb")]
    bb_bbb_bbb: i64,
    #[serde(rename = "ccCccCcc")]
    cc_ccc_ccc: i64,
}
```

This would be more friendly using rename_all instead:

```rust
#[derive(Default, Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
struct ZalandoArticle {
    aa_aaa_aaa: i64,
    bb_bbb_bbb: i64,
    cc_ccc_ccc: i64,
}
```

The supported renames are "lowercase" (which you won't need, mostly for enums), "PascalCase", "camelCase", "snake_case" (also for enum variants), "SCREAMING_SNAKE_CASE", "kebab-case".

**Comment by evestera** [comment]
Thanks!

Yes, I've looked at that `rename_all`, and it does look quite a bit nicer when you actually see the generated code, so I would certainly like to use `rename_all` in the future. Just prioritizing the biggest flaws first (at the moment duplicate type names).

**Comment by evestera** [comment]
Took a while, but finally implemented now in [release 0.4](https://github.com/evestera/json_typegen/releases/tag/v0.4.0) 🎉
